src/test/ui/macros/expr-stmt-nonterminal-tokens.rs

```diff
@@ -0,0 +1,15 @@
+// check-pass
+
+macro_rules! mac {
```

Comment:
Can you modify this test (or add a test) to expand to `#[derive]` target, which verifies that the tokens from the nonterminals are cfg-expanded as expected. E.g.:

```rust
macro_rules! mac {
    (expr $expr:expr) => {
        #[derive(Print)]
        struct Foo {
            field: [bool; { let  = $expr; 0 }]
        }
    }
}
```